src/navigations/RootNavigator.tsx

```diff
@@ -3,12 +3,17 @@ import { createStackNavigator } from '@react-navigation/stack';
 
 import HomeScreen from 'screens/home';
 
-const Stack = createStackNavigator();
+// You can define a type for the stack navigator, so that all the screens and 
+// parameters can be typesafe.
+type RootStackParams = {
+  Home: undefined // Home does not require any special parameter
+}
+const Stack = createStackNavigator<RootStackParams>();
```

Comment:
An example for a case where navigation parameter is specified:
```typescript
type RootStackParams = {
  ViewGif: { giphyId: string }
}
```
Now whoever wants to push the screen named `ViewGif`, they must specify the Giphy ID for the gif image (if not the compiler will complain). This corresponds to the fact that we cannot view a gif without an ID (required parameter for the existence of that screen)